scipy/stats/_continuous_distns.py

```diff
@@ -5550,16 +5550,26 @@ class nakagami_gen(rv_continuous):
 
     """
     def _pdf(self, x, nu):
+        return np.exp(self._logpdf(x, nu))
+
+    def _logpdf(self, x, nu):
         # nakagami.pdf(x, nu) = 2 * nu**nu / gamma(nu) *
         #                       x**(2*nu-1) * exp(-nu*x**2)
-        return 2*nu**nu/sc.gamma(nu)*(x**(2*nu-1.0))*np.exp(-nu*x*x)
+        return (np.log(2) + nu*np.log(nu) - sc.gammaln(nu) +
+                (2*nu - 1)*np.log(x) - nu*x**2)
```

Comment:
Nit, optional: `xlogy(2*nu -1, x)`

Reply:
OK, I've changed that, and I've changed `nu*np.log(nu)` to `xlogy(nu, nu)`.